qiskit/providers/ibmq/managed/managedjob.py

```diff
@@ -95,22 +100,42 @@ def _async_submit(
             qobj: Qobj to run.
             job_name: Name of the job.
             backend: Backend to execute the experiments on.
+            submit_lock: Lock used to synchronize job submission.
             job_share_level: Job share level.
-            job_tags: tags to be assigned to the job.
-
-        Returns:
-            IBMQJob instance for the job.
+            job_tags: Tags to be assigned to the job.
         """
+        # pylint: disable=missing-raises-doc
+        submit_lock.acquire()
+        try_again = True
         try:
-            self.job = backend.run(
-                qobj=qobj,
-                job_name=job_name,
-                job_share_level=job_share_level.value,
-                job_tags=job_tags)
+            while try_again:
+                try_again = False
+                try:
+                    self.job = backend.run(
+                        qobj=qobj,
+                        job_name=job_name,
+                        job_share_level=job_share_level.value,
+                        job_tags=job_tags)
+                except IBMQBackendApiError as api_err:
+                    if 'Error code: 3458' in str(api_err):
+                        logger.warning("Job limit reached, waiting for jobs to finish "
+                                       "before submitting the next one.")
+                        final_states = [state.value for state in API_JOB_FINAL_STATES]
+                        oldest_running = backend.jobs(limit=1, descending=False,
+                                                      db_filter={"status": {"nin": final_states}})
+                        if oldest_running:
+                            oldest_running = oldest_running[0]
+                            logger.info("Waiting for job %s to finish", oldest_running.job_id())
+                            oldest_running.wait_for_final_state(timeout=300)
+                        try_again = True
+                    else:
+                        raise
         except Exception as err:  # pylint: disable=broad-except
             warnings.warn("Unable to submit job for experiments {}-{}: {}".format(
                 self.start_index, self.end_index, err))
             self.submit_error = err
+        finally:
+            submit_lock.release()
```

Comment:
This could maybe be condensed as:
```
try:
    while not self.job:
        try:
            self.job = backend.run(...)
        except IBMQBackendApiError as api_error:
            ...
except Exception as err:
    ...
finally:
    ....
```
This would possibly allow not needing the `try_again` variable?